src/dataset-builder/DomainCriteriaSearch.ts

```diff
@@ -0,0 +1,43 @@
+import { h } from 'react-hyperscript-helpers';
+import { saveSelected } from 'src/dataset-builder/DomainCriteriaSelector';
+import { SnapshotBuilderConcept as Concept, SnapshotBuilderDomainOption as DomainOption } from 'src/libs/ajax/DataRepo';
+
+import { ConceptSearch } from './ConceptSearch';
+import {
+  cohortEditorState,
+  DomainCriteriaSearchState,
+  domainCriteriaSearchState,
+  domainCriteriaSelectorState,
+} from './dataset-builder-types';
+import { OnStateChangeHandler } from './DatasetBuilder';
+
+interface DomainCriteriaSearchProps {
+  readonly state: DomainCriteriaSearchState;
+  readonly onStateChange: OnStateChangeHandler;
+  readonly datasetId: string;
+  readonly getNextCriteriaIndex: () => number;
+}
+
+export const DomainCriteriaSearch = (props: DomainCriteriaSearchProps) => {
+  const { state, onStateChange, datasetId, getNextCriteriaIndex } = props;
+  return h(ConceptSearch, {
+    initialSearch: state.searchText,
+    initialCart: state.cart,
+    domainOption: state.domainOption,
+    onCancel: () => onStateChange(cohortEditorState.new(state.cohort)),
+    onCommit: saveSelected(state, getNextCriteriaIndex, onStateChange),
+    onOpenHierarchy: (domainOption: DomainOption, cart: Concept[], searchText: string) => {
+      onStateChange(
+        domainCriteriaSelectorState.new(
+          state.cohort,
+          state.criteriaGroup,
+          domainOption,
+          cart,
+          domainCriteriaSearchState.new(state.cohort, state.criteriaGroup, state.domainOption, cart, searchText)
+        )
+      );
+    },
+    actionText: 'Add to group',
```

Comment:
I think what you're saying is, you'd expect the group number as shown to be numbered based on the visible group elements rather than a canonical number as assigned by the UI. That makes sense as the UI's group numbers aren't useful to a user. This was probably the intent of the original UX screens but we lost that when working on the code, which requires a unique ID internally.

I'll make a ticket to look into this and fix it. I'm not completely sure the UI needs a number here at all, which is another option.

Reply:
Ok!